spec/services/idv/profile_step_spec.rb

```diff
@@ -19,93 +19,111 @@
+      expect(step.submit).to eq result
+      expect(idv_session.profile_confirmation).to eq false
+    end
+
+    it 'fails when form validation fails' do
+      step = build_step(user_attrs.merge(ssn: '6666'))
+
+      errors = { ssn: [t('idv.errors.pattern_mismatch.ssn')] }
+      extra = {
+        idv_attempts_exceeded: false,
+        vendor: { reasons: nil }
+      }
+
+      result = instance_double(FormResponse)
+
+      expect(FormResponse).to receive(:new).
+        with(success: false, errors: errors, extra: extra).and_return(result)
+      expect(step.submit).to eq result
+      expect(idv_session.profile_confirmation).to eq false
     end
 
     it 'fails with invalid first name' do
       step = build_step(user_attrs.merge(first_name: 'Bad'))
 
-      expect(step.complete).to eq false
-      expect(step.complete?).to eq false
+      errors = { first_name: ['Unverified first name.'] }
 
-      result = {
-        success: false,
+      result = instance_double(FormResponse)
+      extra = {
         idv_attempts_exceeded: false,
-        errors: {
-          first_name: ['Unverified first name.']
-        },
         vendor: { reasons: ['The name was suspicious'] }
       }
 
-      expect_analytics_result(result)
+      expect(FormResponse).to receive(:new).
+        with(success: false, errors: errors, extra: extra).and_return(result)
+      expect(step.submit).to eq result
+      expect(idv_session.profile_confirmation).to eq false
     end
-  end
 
-  describe '#attempts_exceeded?' do
-    it 'tracks resolution attempts' do
-      user.idv_attempts = 3
-      user.idv_attempted_at = Time.zone.now
+    it 'increments attempts count if the form is valid' do
       step = build_step(user_attrs)
+      expect { step.submit }.to change(user, :idv_attempts).by(1)
+    end
 
-      step.complete
-      expect(step.attempts_exceeded?).to eq true
+    it 'does not increment the attempts count if the form is not valid' do
+      step = build_step(user_attrs.merge(ssn: '666'))
+      expect { step.submit }.to change(user, :idv_attempts).by(0)
+    end
 
-      result = {
-        success: false,
-        idv_attempts_exceeded: true,
-        errors: {},
-        vendor: { reasons: nil }
-      }
+    it 'initializes the idv_session' do
+      step = build_step(user_attrs)
+      step.submit
 
-      expect_analytics_result(result)
+      expect(idv_session.params).to eq user_attrs
+      expect(idv_session.applicant).to be_a Proofer::Applicant
+      user_attrs.each do |k, v|
+        expect(idv_session.applicant.send(k)).to eq v
+      end
+    end
+  end
+
+  describe '#attempts_exceeded?' do
+    it 'calls Idv::Attempter#exceeded?' do
+      attempter = instance_double(Idv::Attempter)
+      allow(Idv::Attempter).to receive(:new).with(user).and_return(attempter)
+      allow(attempter).to receive(:exceeded?)
+
+      step = build_step(user_attrs)
+      expect(step.attempts_exceeded?).to eq attempter.exceeded?
```

Comment:
It already does implicitly. The test will fail if `attempter` did not receive `exceeded?`. I wanted to ensure that the return value was exactly `attempter.exceeded?` without having to write the test for `exceeded?` again since it's already tested in the `Idv::Attempter` spec. If we only checked that it received the message, it wouldn't be enough because something else could happen after the call which could alter the return value.

On the other hand, this could be considered testing implementation, so perhaps we should just test the return value given an input, i.e. set the user's attempts to above and below the threshhold, and test for `true` and `false` return values, which would somewhat duplicate the existing test for `Idv::Attempter#exceeded?`, but this kind of duplication is OK I guess.